### DISCUSSION TITLE
Error adding helia dependency

### DISCUSSION THREAD

**Discussion by jeluard** [parcel-bundler/parcel#9077] [General] [answered]
Hi!

Trying to add [helia](https://github.com/ipfs/helia) (`yarn add helia`) then importing it triggers the following parcel error:

```js
🚨 Build failed.

@parcel/core: Failed to resolve 'helia' from './src/app.js'

/repros/parcel-helia/src/app.js:1:29
  > 1 | import { createHelia } from 'helia'
  >   |                             ^^^^^^^
    2 |
    3 | console.log("Hello");

@parcel/resolver-default: Cannot load file './index' from module 'helia'
✨  Done in 8.46s.
```

Am I missing something obvious? In works fine with e.g. `esbuild`

**Comment by mischnic** [comment, marked as answer]
That package has only an `exports` field, no `main` or some other legacy field: https://github.com/ipfs/helia/blob/671ec874e90fbdcaf79d9d8253822fd85cee8bc5/packages/helia/package.json#L29

https://parceljs.org/blog/v2-9-0/#new-resolver:
> Unfortunately, enabling support for package.json "exports" is a breaking change. When a package declares "exports", consumers can no longer import files that are not exported. In addition, due to exports conditions like "import" and "require", there is a possibility of a [dual package hazard](https://nodejs.org/api/packages.html#dual-package-hazard). For these reasons, exports support is currently opt-in and must be explicitly enabled in your project. This can be done by adding the following to your project root package.json:
> ```json
>{
>  "@parcel/resolver-default": {
>    "packageExports": true
>  }
>}
>```

**Reply by jeluard** [reply]
Thanks, I missed that one!

**Reply by jeluard** [reply]
@mischnic Out of curiosity, what is the longer term plan here? Am I right that packages like `helia` do the right thing, and that bundlers will eventually support them natively?

**Reply by mischnic** [reply]
We would really like to enable exports support by default, but it's simply a breaking change. So it'll have to wait until we release a major version. So eventually yes...
